src/components/inputs/date-input/date-input.spec.js

```diff
@@ -44,6 +45,29 @@ class Story extends React.Component {
 const renderDateInput = (props, options) =>
   render(<Story {...props} />, options);
 
+describe('interaction', () => {
+  describe('interacting with header buttons', () => {
+    it('should not blur the input', async () => {
+      const onFocus = jest.fn();
+      const onBlur = jest.fn();
+      const { getByLabelText, getByText, queryByText } = renderDateInput({
+        value: '2018-09-18',
+        onFocus,
+        onBlur,
+      });
+      const dateInput = getByLabelText('Date');
+      userEvent.click(dateInput);
+      expect(onFocus).toHaveBeenCalledTimes(1);
+      expect(getByText('September')).toBeInTheDocument();
+      const previousMonthButton = getByLabelText('show prev month');
+      userEvent.click(previousMonthButton);
```

Comment:
Would you suggest using this in the MC too or apply it everywhere in UIKit?

Reply:
@tdeekens yes. Now that I realize that `fireEvent.click` doesn't actually ... click, I think it's the worst and should be deleted.